src/components/pages/Dashboard/index.tsx

```diff
@@ -1,38 +1,105 @@
 import * as React from 'react';
-import { retrieveProfile } from '../../../store/actions/dashboard';
 import { connect } from 'react-redux';
-import withAuthenticationContainer from '../../containers/AuthenticationContainer';
+import { Switch, Route, match, RouteProps } from 'react-router';
+import { Location } from 'history';
+
+import withSuspense from 'components/HOC/withSuspense';
+import withDashboardContainer from 'components/containers/DashBoardContainer';
+import { retrieveProfile } from 'store/actions/dashboard';
+import DashBoardLayout from 'components/layouts/DashBoardLayout';
 
 interface DashboardProps {
   profile: {
     firstName: string;
     lastName: string;
     username: string;
+    email: string;
     verified: string;
+    imageURL: string;
   };
   retrieveProfile: () => void;
+  match: match<{}>;
+  location: Location;
+}
+
+function getDashBoardRoutes(path: string): RouteProps[] {
+  return [
+    {
+      path: `${path}`,
+      component: withSuspense({ page: 'MyOrganisations' }),
+      exact: true,
+    },
+    {
+      path: `${path}/profile`,
+      component: withSuspense({ page: 'UpdateProfile' }),
+      exact: true,
+    },
+    {
+      path: `${path}/organisation`,
+      component: withSuspense({ page: 'CreateOrganisation' }),
+      exact: true,
+    },
+    {
+      path: `${path}/password`,
+      component: withSuspense({ page: 'UpdatePassword' }),
+      exact: true,
+    },
+  ];
 }
-function Dashboard({ profile, retrieveProfile }: DashboardProps) {
-  React.useEffect(() => {
+
+class Dashboard extends React.Component<DashboardProps, {}> {
```

Comment:
Is there a way to avoid using a class-based component here. Looks kinda off since we have been using function-based